### PR TITLE
VACMS-12709: Commit enable TMGMT and set permissions.

### LINKED ISSUE #12709: Enable TMGMT and associated modules
## Description
[TMGMT](https://www.drupal.org/project/tmgmt) is the defacto standard for managing the translation process in Drupal. We will be using it for VA.gov. 

## Acceptance criteria
- [ ] TMGMT is installed.
- [ ] XLIFF files containing translation data can be exported and imported.

### PR DESCRIPTION
**Pull request by timcosgrove**
## Description

Closes #12709 and #12710

## Testing done
- Tests run
- Run export/import successfully

## QA steps
What needs to be checked to prove this works?  

As user with the `translation_manager` role (may need to add the role to an existing user or create a new test user)
1. Visit sources page (https://pr13021-fn26fgyjzcrw4n3fiuz6upy4fonk0cla.ci.cms.va.gov/admin/tmgmt/sources). Filter to Support Services nodes (these are very short so good for testing). Check a few nodes, select English as the source and Spanish as the target. 'Request translation'. You will be taken to a new form. When you open up the Job items in the upper right, you see the various nodes you added in there.
   - [x] The items are visible in Job items
2. Provide a label for the job if you like. Click 'Submit to provider'. You will be taken back to the list of Sources. There will be an alert up top that says 'Exported file can be downloaded [here](http://va-gov-cms.ddev.site/sites/default/files/tmgmt_file/JobID5_en_es.xlf).' Confirm you can download the file. **Note:** you may get a download error due to security issues. You can either bypass the warning your browser gives you, or load the XLIFF file into a new tab and then save it to your machine. This will not be an issue in staging or prod.
   - [x] The file can be downloaded.
3. Download an XLIFF editor like Counterparts Lite. (https://michelf.ca/software/counterparts-lite/). Use this editor to enter test content for the file. Save it.
4. Go to the Jobs page ([/admin/tmgmt/jobs](https://pr13021-fn26fgyjzcrw4n3fiuz6upy4fonk0cla.ci.cms.va.gov/admin/tmgmt/jobs)). Find your job. Click 'manage'. Scroll to 'import file', add your 'translated' file, and import it.
- [x] The translation file is imported successfully.
5. Go back into 'manage' on your test job. You should see see your job items with yellow icons/colors labeling them, indicating they need to be reviewed. Review each item, and save as completed.
   - [x] Each item can be reviewed
   - [x] The translation matches what was entered in step 3.
   - [x] The translation can be saved as completed.
6. Go to the content admin (https://pr13021-fn26fgyjzcrw4n3fiuz6upy4fonk0cla.ci.cms.va.gov/admin/content). Filter to Support Service.
  - [x] You should see the nodes that you translated.
  - [x] If you visit or edit the translations, they should contain the translated content you provided in 3. 

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

File: config/sync/core.entity_form_display.user.user.default.yml [hunk 1]
```diff
@@ -7,9 +7,11 @@ dependencies:
     - field.field.user.user.field_password_expiration
     - field.field.user.user.field_pending_expire_sent
     - field.field.user.user.message_subscribe_email
+    - field.field.user.user.tmgmt_translation_skills
```
Comment (timcosgrove) [inline config/sync/core.entity_form_display.user.user.default.yml:10]: The additions to this form regulate what users will be allowed to 'offer' translation from one language to another via TMGMT. This is related to the 'Drupal user' translation provider. We may well remove this, since most if not all translation will be managed via file export and import, but, wanted to include the option for the time being.

File: config/sync/language.types.yml [hunk 2]
```diff
@@ -16,4 +16,5 @@ negotiation:
       language-url-fallback: 1
   language_interface:
     enabled:
+      language-graphql: -999
```
Comment (timcosgrove) [inline config/sync/language.types.yml:19]: Graphql installs this; this determines what language Drupal understands it should return translations for. Per this language negotiation method:
>  The current GraphQL language context. Only available while executing a query.

See /admin/config/regional/language/detection if interested.

File: config/sync/tmgmt.settings.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+_core:
+  default_config_hash: kNAclv7Nq8cW0Huko5DpTk2YAMaXNiCVB9iZON_MNHc
+quick_checkout: true
+anonymous_access: true
```
Comment (timcosgrove) [inline config/sync/tmgmt.settings.yml:4]: This is not general anonymous access; it provides Drupal access to unpublished/draft content items for translation managers who might otherwise not be assigned access to draft content. It basically enables the translation manager to add unpublished content to their translation jobs.

File: config/sync/tmgmt_content.settings.yml [hunk 4]
```diff
@@ -0,0 +1,5 @@
+_core:
+  default_config_hash: zJBQPBc6SKaZZvDCvMKeZm7_0CH-7OCtq92DRu-vuPs
+embedded_fields: {  }
+default_moderation_states:
+  editorial: draft
```
Comment (timcosgrove) [inline config/sync/tmgmt_content.settings.yml:5]: For the time being, translations will come in as draft content regardless of the moderation state of the source. We may changes this later; the other option is to have an accepted translation immediately match the moderation state of the source.

File: config/sync/language/es/system.action.tmgmt_local_task_unassign_multiple.yml [hunk 5]
```diff
@@ -0,0 +1 @@
+label: Desasignar
```
Comment (timcosgrove) [inline config/sync/language/es/system.action.tmgmt_local_task_unassign_multiple.yml:1]: These and following simply provide Spanish translation for views etc when the site is being 'viewed' in Spanish. These configuration translations will most likely be removed.

File: tests/phpunit/Security/RolesPermissionsTest.php [hunk 6]
```diff
@@ -414,8 +416,8 @@ public function expectedPerms() {
           'view vet_center_mobile_vet_center revisions',
           'view vet_center_outstation revisions',
           'view vha_facility_nonclinical_service revisions',
-          'view vha_facility_nonclinical_service revisions',
           'view video revisions',
+          'view vha_facility_nonclinical_service revisions',
```
Comment (ndouglas) [inline tests/phpunit/Security/RolesPermissionsTest.php:420]: This change shouldn't be required after the recent sorting PR, right?

**Review by ndouglas — APPROVED** [review]
Aight, so I created a test account:

Screenshot 2023-04-04 at 7 26 01 AM

Poked around in a couple of nodes.

**Question**: Is this accessible? Is this appearing as intended?

Screenshot 2023-04-04 at 7 28 40 AM
Screenshot 2023-04-04 at 7 28 50 AM
Screenshot 2023-04-04 at 7 29 56 AM

**Comment**: Just click "Request translation" without selecting a language.  This is janky -- maybe there's an open issue for this upstream.  Not connected to your PR, but I wanna complain anyway.

Screenshot 2023-04-04 at 7 29 21 AM

**Comment**: This is a _Core_ module, you say?  I don't like this, and I don't know if it's just because the interface is unfamiliar or because it is not good.

Screenshot 2023-04-04 at 7 31 49 AM

Job item shows up as expected.
Screenshot 2023-04-04 at 7 32 42 AM

Translation requested as expected.
Screenshot 2023-04-04 at 7 32 56 AM

¡No parle Spangläiçe!
Screenshot 2023-04-04 at 7 36 13 AM

Import worked.
Screenshot 2023-04-04 at 7 35 45 AM
Screenshot 2023-04-04 at 7 36 20 AM
Screenshot 2023-04-04 at 7 37 43 AM

**Question**: Seems... unnecessary to display this message.  Right?  Or am I misunderstanding?  I was just viewing the page.

Screenshot 2023-04-04 at 7 37 51 AM

**Question**: If I click the Spanish translation, I still see text in English. It appears to be because the translation is in Draft. Is this intended? Is this intuitive?

Screenshot 2023-04-04 at 7 38 11 AM

If I publish the translation draft, then I see the "Spanish" text.

Screenshot 2023-04-04 at 7 51 55 AM

But the warning about fields seems to be displayed at the wrong time. It's shown on the edit form _and_ when saved.  Again, I don't think this is related to your work, but it's not really giving me a great impression of the core system.

Now, if I view the most recent revision of the translation, I see this:

Screenshot 2023-04-04 at 7 54 53 AM

**Suggestion (nonblocking)**: Issue is that we should probably display the sidebar with revision history here, I think.  Do you agree?  That's negotiable and fine as a followup.

**TL;DR**: I don't see any showstoppers here, given that the whole translation effort is a work in progress.  I feel like there are a lot of things that require followup here, though, to get the translation system to behave well.  Not sure if I'm being rational or not.

File: tests/phpunit/Security/RolesPermissionsTest.php (hunk 6 above)
Comment (timcosgrove) [reply tests/phpunit/Security/RolesPermissionsTest.php:420]: True. I've been trying to get this PR merged for ~2 weeks :) 

I merged your sorting PR specifically to make this PR easier to keep up-to-date.
